.github/workflows/testing.yml

```diff
@@ -9,9 +9,82 @@ on:
     tags:
     - v*
   pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
   workflow_dispatch:
 
 jobs:
+  skips-allowed:
+    name: Determine CI Skipping
+    runs-on: ubuntu-latest
+
+    env:
+      IS_PR: ${{ github.event_name == 'pull_request' }}
+      SKIP_DOCUMENTATION_LABEL: 'GA Skip Documentation'
+      SKIP_COMPREHENSIVE_LABEL: 'GA Skip Comprehensive Tests'
+      SKIPS_ALLOWED_LABEL: 'GA Skip Merge Allowed'
+
+      SKIP_DOCUMENTATION: false
+      SKIP_COMPREHENSIVE: false
+      SKIPS_ALLOWED: false
+
+      SKIPS_DONE: false
+
+    outputs:
+      SKIP_DOCUMENTATION: ${{ env.SKIP_DOCUMENTATION }}
+      SKIP_COMPREHENSIVE: ${{ env.SKIP_COMPREHENSIVE }}
+      SKIPS_ALLOWED: ${{ env.SKIPS_ALLOWED }}
+      SKIPS_DONE: ${{ env.SKIPS_DONE }}
+
+    steps:
+      - name: Determine Skips
+        id: step1
+        if: ${{ env.IS_PR }}
+        run: |
+          SKIP=${{ contains(github.event.pull_request.labels.*.name, env.SKIP_DOCUMENTATION_LABEL) }}
+          echo "SKIP_DOCUMENTATION=${SKIP}" >> $GITHUB_ENV
+
+          SKIP=${{ contains(github.event.pull_request.labels.*.name, env.SKIP_COMPREHENSIVE_LABEL) }}
+          echo "SKIP_COMPREHENSIVE=${SKIP}" >> $GITHUB_ENV
+
+          SKIP=${{ contains(github.event.pull_request.labels.*.name, env.SKIPS_ALLOWED_LABEL) }}
+          echo "SKIPS_ALLOWED=${SKIP}" >> $GITHUB_ENV
+
+      - name: Skip Documentation? ${{ env.SKIP_DOCUMENTATION }}
+        run: |
+          echo "Documentation can be skipped: ${{ env.SKIP_DOCUMENTATION }}"
+          if ${{ env.SKIP_DOCUMENTATION }}; then
+            echo "SKIPS_DONE=true" >> $GITHUB_ENV
+          fi
+      - name: Skip Comprehensive Tests? ${{ env.SKIP_COMPREHENSIVE }}
+        run: |
+          echo "Comprehensive tests can be skipped: ${{ env.SKIP_COMPREHENSIVE }}"
+          if ${{ env.SKIP_COMPREHENSIVE }}; then
+            echo "SKIPS_DONE=true" >> $GITHUB_ENV
+          fi
+
+  mergeble-with-skips:
+    name: Skipping and Merge Allowed?
+    runs-on: ubuntu-latest
+    needs: skips-allowed
+
+    env:
+      SKIPS_ALLOWED: ${{ needs.skips-allowed.outputs.SKIPS_ALLOWED }}
+      SKIPS_DONE: ${{ needs.skips-allowed.outputs.SKIPS_DONE }}
+
+    steps:
+      - name: Skip Tests Allowed? ${{ env.SKIPS_ALLOWED }}
+        run: |
+          echo "Is skipping tests allowed: ${{ env.SKIPS_ALLOWED }}"
+          if [[ "$SKIPS_ALLOWED" == "false" && "$SKIPS_DONE" == "true" ]]; then
+            echo "Some tests are skipped, but skipping needs to be removed before merging!!"
```

Comment:
I believe this would work.  I originally didn't write something like this because I only wanted us to know about the "skips allows" functionality.  Only we should be adding that label and not the common contributor.